src/coreclr/vm/genmeth.cpp

```diff
@@ -1267,8 +1267,7 @@ MethodDesc::FindOrCreateAssociatedMethodDesc(MethodDesc* pDefMD,
         // that keeps the reflection logic the same for value types
 
         // we need unboxing stubs for virtual methods on value types unless the method is generic
-        BOOL fNeedUnboxingStub = pMethod->IsUnboxingStub() ||
-            ( instType.IsValueType() && pMethod->IsVirtual() );
+        BOOL fNeedUnboxingStub = instType.IsValueType() && pMethod->IsVirtual();
```

Comment:
Similar `pMethod->IsUnboxingStub(),` call is done above (line 1244). Should fix up that one as well?

Reply:
Right. That is the one of the instances I wasn't sure if we needed to update too.

Reply:
By my understanding that is the case of instantiated case for which I would expect that `pMethod->IsUnboxingStub()` is already set as needed

Reply:
And the comment above it matches with the behavior: https://github.com/dotnet/runtime/blob/7015fa2d28e4559478807a53b5851c71d2f604e8/src/coreclr/vm/genmeth.cpp#L1239-L1244

Reply:
Learnt that this path is for generic method and I seem similar issue (i.e. different handle for the 2 case), so yes its need update and updated, thanks!